### PR TITLE
ESQ: Add metadata fields documentation

### PR DESCRIPTION
**Pull request by bpintea**
This adds documentation for the supported metadata fields.

### CODE REVIEW

File: docs/reference/esql/metadata-fields.asciidoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+[[esql-metadata-fields]]
+== {esql} metadata fields
+
+++++
+<titleabbrev>Metadata fields</titleabbrev>
+++++
+
+{esql} can also access <<mapping-fields, metadata fields>>. The currently
```
Comment (abdonpijpelink) [inline docs/reference/esql/metadata-fields.asciidoc:8]:
```suggestion
{esql} can access <<mapping-fields, metadata fields>>. The currently
```

File: docs/reference/esql/metadata-fields.asciidoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  * <<mapping-index-field, _index>>: the index to which the document belongs.
+  The field is of the type <<keyword, keyword>>.
+
+  * <<mapping-id-field, _id>>: the source document's ID. The field is of the
+  type <<keyword, keyword>>.
+
+  * _version: the source document's version. The field is of the type
+  <<number, long>>.
+
+To enable the access to these fields, the <<esql-from, from>> source command needs
```
Comment (abdonpijpelink) [inline docs/reference/esql/metadata-fields.asciidoc:20]:
```suggestion
To enable the access to these fields, the <<esql-from,`FROM`>> source command needs
```

File: docs/reference/esql/metadata-fields.asciidoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+supported ones are:
+
+  * <<mapping-index-field, _index>>: the index to which the document belongs.
+  The field is of the type <<keyword, keyword>>.
+
+  * <<mapping-id-field, _id>>: the source document's ID. The field is of the
+  type <<keyword, keyword>>.
+
+  * _version: the source document's version. The field is of the type
+  <<number, long>>.
```
Comment (abdonpijpelink) [inline docs/reference/esql/metadata-fields.asciidoc:18]:
```suggestion
  * <<mapping-index-field,`_index`>>: the index to which the document belongs.
  The field is of the type <<keyword, keyword>>.

  * <<mapping-id-field,`_id`>>: the source document's ID. The field is of the
  type <<keyword, keyword>>.

  * `_version`: the source document's version. The field is of the type
  <<number,long>>.
```

File: docs/reference/esql/metadata-fields.asciidoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+To enable the access to these fields, the <<esql-from, from>> source command needs
+to be provided with a dedicated directive:
+
+[source,esql]
+----
+FROM index [METADATA _index, _id]
+----
+
+Consequently, the metadata fields are only available if the source of the data
+is an index; i.e., the source command is `FROM`.
```
Comment (abdonpijpelink) [inline docs/reference/esql/metadata-fields.asciidoc:29]:
```suggestion
Metadata fields are only available if the source of the data is an index.
Consequently, `FROM` is the only source commands that supports the `METADATA`
directive.
```

Comment (abdonpijpelink) [reply docs/reference/esql/metadata-fields.asciidoc:29]: Just a suggestion.

File: docs/reference/esql/metadata-fields.asciidoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+----
+include::{esql-specs}/metadata-ignoreCsvTests.csv-spec[tag=multipleIndices]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/metadata-ignoreCsvTests.csv-spec[tag=multipleIndices-result]
+|===
+
+Also, similar to the index fields, once an aggregation is performed, a
+metadata field will no longer be "accessible" to subsequent commands, unless
```
Comment (abdonpijpelink) [inline docs/reference/esql/metadata-fields.asciidoc:44]:
```suggestion
metadata field will no longer be accessible to subsequent commands, unless
```

File: docs/reference/esql/source-commands/from.asciidoc [hunk 6]
```diff
@@ -27,3 +27,10 @@ or aliases:
 ----
 FROM employees-00001,employees-*
 ----
+
+Use the `metadata` directive to enable <<esql-metadata-fields, metadata fields>>:
```
Comment (abdonpijpelink) [inline docs/reference/esql/source-commands/from.asciidoc:31]:
```suggestion
Use the `METADATA` directive to enable <<esql-metadata-fields,metadata fields>>:
```

**Review by abdonpijpelink — APPROVED** [review]
Looks good. I've left a few suggestions.

**Comment by bpintea** [comment]
Thanks, Abdon!

**Pull request merged**